src/prommis/uky/costing/ree_plant_capcost.py

```diff
@@ -2669,3 +2858,737 @@ def costing_upper_bound_eq(c, p):
+                        pyunits.convert(
+                            c.discount_percentage, to_units=pyunits.dimensionless
+                        ),
+                        pyunits.convert(
+                            c.operating_inflation_percentage,
+                            to_units=pyunits.dimensionless,
+                        ),
+                        len(c.config.capital_expenditure_percentages),
+                    )
+                ),
+                to_units=c.cost_units,
+            )
+
+        @b.Constraint()
+        def pv_revenue_constraint(c):
+            # PV_Revenue = REVENUE * [ P/A_OPEX+CAPEX_periods - P/A_CAPEX_period ]
+
+            return c.pv_revenue == pyunits.convert(
+                c.REVENUE[None]
+                * (
+                    series_present_worth_factor(
+                        pyunits.convert(
+                            c.discount_percentage, to_units=pyunits.dimensionless
+                        ),
+                        pyunits.convert(
+                            c.revenue_inflation_percentage,
+                            to_units=pyunits.dimensionless,
+                        ),
+                        c.plant_lifetime / pyunits.year
+                        + len(c.config.capital_expenditure_percentages),
+                    )
+                    - series_present_worth_factor(
+                        pyunits.convert(
+                            c.discount_percentage, to_units=pyunits.dimensionless
+                        ),
+                        pyunits.convert(
+                            c.revenue_inflation_percentage,
+                            to_units=pyunits.dimensionless,
+                        ),
+                        len(c.config.capital_expenditure_percentages),
+                    )
+                ),
+                to_units=c.cost_units,
+            )
+
+        if b.config.royalty_expression is None:
+            # PV_Royalties = - %royalty_charge_of_revenue * REVENUE
+
+            @b.Constraint()
+            def pv_royalties_constraint(c):
+
+                return c.pv_royalties == -pyunits.convert(
+                    pyunits.convert(
+                        c.royalty_charge_percentage_of_revenue,
+                        to_units=pyunits.dimensionless,
+                    )
+                    * c.pv_revenue,
+                    to_units=c.cost_units,
+                )
+
+        else:
+
+            b.pv_royalties = Reference(b.config.royalty_expression)
+
+        if b.config.debt_expression is None:
+
+            @b.Constraint()
+            def loan_debt_constraint(c):
+                # Debt  = - %debt_charge_of_CAPEX * CAPEX
+
+                return c.loan_debt == pyunits.convert(
+                    pyunits.convert(
+                        c.debt_percentage_of_CAPEX, to_units=pyunits.dimensionless
+                    )
+                    * c.CAPEX,
+                    to_units=c.cost_units,
+                )
```

Comment:
Should this equation have a negative sign in front?

Reply:
```suggestion
            def loan_debt_constraint(c):
                # Debt  = %debt_charge_of_CAPEX * CAPEX

                return c.loan_debt == pyunits.convert(
                    pyunits.convert(
                        c.debt_percentage_of_CAPEX, to_units=pyunits.dimensionless
                    )
                    * c.CAPEX,
                    to_units=c.cost_units,
                )
```

The equation is correct, but there shouldn't be a negative sign in the comment. The debt is taken as a positive value so that the interest owed (a negative cash flow) is `interest_owed = - (payment_per_year * number_of_years - debt)`.